be/src/exprs/aggregate_functions.cpp

```diff
@@ -231,6 +232,84 @@ void AggregateFunctions::avg_update(FunctionContext* ctx, const T& src, StringVa
     ++avg->count;
 }
 
+struct PercentileApproxState {
+public:
+    TDigest *digest = nullptr;
+    double targetQuantile = -1.0;
+
+    ~PercentileApproxState() {
+        if (digest != nullptr) {
+            delete digest;
+        }
+    }
+};
+
+void AggregateFunctions::percentile_approx_init(FunctionContext* ctx, StringVal* dst) {
+    dst->is_null = false;
+    dst->len = sizeof(PercentileApproxState);
+    dst->ptr = (uint8_t*) new PercentileApproxState();
+
+    PercentileApproxState* percentile = reinterpret_cast<PercentileApproxState*>(dst->ptr);
+    percentile->targetQuantile = -1.0;
+    percentile->digest = new TDigest();
+};
+
+template<typename T>
+void AggregateFunctions::percentile_approx_update(FunctionContext* ctx, const T& src, const DoubleVal& quantile, StringVal* dst) {
+    if (src.is_null) {
+        return;
+    }
+    DCHECK(dst->ptr != NULL);
+    DCHECK_EQ(sizeof(PercentileApproxState), dst->len);
+
+    PercentileApproxState* percentile = reinterpret_cast<PercentileApproxState*>(dst->ptr);
+    percentile->digest->add(src.val);
+    percentile->targetQuantile = quantile.val;
```

Comment:
I think you should check quantile in Frontend which should be a const expression. If quantile is not const, I think this function has no use.

Reply:
done